lib/AST/ASTImporter.cpp

```diff
@@ -6806,31 +6804,32 @@ Decl *ASTImporter::GetVAListTag(Decl *FromD) {
 }
 
 Decl *ASTImporter::Import(Decl *FromD) {
+
   if (!FromD)
     return nullptr;
 
+  if (VisitedDecls.count(FromD) > 0)
+    return GetAlreadyImportedOrNull(FromD);
```

Comment:
Probably assert here instead of returning `nullptr` in not-found case?

Reply:
Not good, in case of the templated vs described import cycle we may return with a nullptr.
E.g. during a FunctionTemplateDecl import we have to first import the templated decl, but during import of the template decl we have to import the template itself. The second time when we call the VisitFunctionTemplateDecl we return with a nullptr, since we already visited it but we don't have an imported definition to that (yet). 

Perhaps using an optional<Decl*> would be more appropriate, so we could distinguish error from the above situation. Or we could have a global pointer constant which has this meaning. @Xazax-hun what do you think?

Reply:
Probably assert in every case of error and return null pointer only if the node is under import but not yet created?